src/peer.cxx

```diff
@@ -231,6 +240,8 @@ bool peer::recreate_rpc(ptr<srv_config>& config,
         //   hence reset timer.
         reset_active_timer();
 
+        reset_stream();
```

Comment:
This should also be called in `handle_rpc_result`, when `err` exists.

Reply:
How can we handle the rejected request in stream mode? It is not an error.

Reply:
I see that we call set_free() here. Don't we need to call reset_stream() when the rpc recreated?